### PR TITLE
Add Support for Amazon nova top k (#7087)

### LINKED ISSUE #7087: [Bug]: LiteLLM Not Supporting `top_k` or `topK` input parameters for Amazon Nova
### What happened?

LiteLLM is not supporting `top_k` or `topK` for Amazon Nova; repro example here:

```
@pytest.mark.parametrize(
    "model",
    [
        "bedrock/amazon.nova-lite-v1:0",
        "bedrock/amazon.nova-pro-v1:0",
        "bedrock/amazon.nova-micro-v1:0",
    ],
)
def test_nova_tool_calling(model):
    """
    # related issue: https://github.com/BerriAI/litellm/issues/5007
    # Bedrock tool names must satisfy regular expression pattern: [a-zA-Z][a-zA-Z0-9_]* ensure this is true
    """
    #litellm.set_verbose = True
    response = litellm.completion(
        model=model,
        temperature=1.0,
        top_p=1.0,
        top_k=1.0,
        messages=[
            {
                "role": "user",
                "content": "What's the weather like in Boston today in Fahrenheit?",
            }
        ],
        tools=[
            {
                "type": "function",
                "function": {
                    "name": "-DoSomethingVeryCool-forLitellm_Testin999229291-0293993",
                    "description": "use this to get the current weather",
                    "parameters": {"type": "object", "properties": {}},
                },
            }
        ],
    )
```

Failure is shown in logs below:

### Relevant log output

```shell
>                       raise BadRequestError(
                            message=f"BedrockException - {error_str}",
                            model=model,
                            llm_provider="bedrock",
                            response=getattr(original_exception, "response", None),
E                           litellm.exceptions.BadRequestError: litellm.BadRequestError: BedrockException - {"message":"The model returned the following errors: Malformed input request: #: extraneous key [top_k] is not permitted, please reformat your input and try again."}

litellm/litellm_core_utils/exception_mapping_utils.py:758: BadRequestError
```
```


### Are you a ML Ops Team?

No

### What LiteLLM version are you on ?

HEAD

### Twitter / LinkedIn details

_No response_

### PR DESCRIPTION
**Pull request by ryanh-ai**
## Title

Add Support for Amazon Nova Top K

## Relevant issues

Fixes #7087 

## Type
🆕 New Feature
✅ Test

## Changes

- Added support for topK
- Added Associated Validating Test

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locall
CleanShot 2024-12-07 at 18 04 43@2x

### CODE REVIEW

**Comment by ryanh-ai** [comment]
@krrishdholakia let me know if you see changes needed here

File: litellm/llms/bedrock/chat/converse_transformation.py [hunk 1]
```diff
@@ -344,10 +347,15 @@ def _transform_request(
                 additional_request_keys.append(k)
         for key in additional_request_keys:
             inference_params.pop(key, None)
+            # Add Support for Amazon Nova topK: https://docs.aws.amazon.com/nova/latest/userguide/complete-request-schema.html
+            if key == "top_k":
```
Comment (ghost) [inline litellm/llms/bedrock/chat/converse_transformation.py:351]: there is no need for this if condition. Line 346 should handle this.

Comment (ghost) [reply litellm/llms/bedrock/chat/converse_transformation.py:351]: please add the topk mapping here inside `map_openai_params()` - https://github.com/BerriAI/litellm/blob/a0aef3b53a015d4877cd91d74edad468949d4f5e/litellm/llms/bedrock/chat/converse_transformation.py#L230

Comment (ryanh-ai) [reply litellm/llms/bedrock/chat/converse_transformation.py:351]: I tried using map_openai_params and I couldn't get it to move under inferenceConfig object in additionalRequestParams

Let me take another look

Comment (ryanh-ai) [reply litellm/llms/bedrock/chat/converse_transformation.py:351]: > there is no need for this if condition. Line 346 should handle this. 

The topK has to move under inferenceConfig key

I suppose we could move the condition into the block above

Comment (ryanh-ai) [reply litellm/llms/bedrock/chat/converse_transformation.py:351]: > please add the topk mapping here inside `map_openai_params()` - https://github.com/BerriAI/litellm/blob/a0aef3b53a015d4877cd91d74edad468949d4f5e/litellm/llms/bedrock/chat/converse_transformation.py#L230

I tried using map_openai_params and I couldn't get it to move under inferenceConfig object in additionalRequestParams

Let me take another look

Comment (ryanh-ai) [reply litellm/llms/bedrock/chat/converse_transformation.py:351]: @krrishdholakia I tested your PR and it isn't setting `topK` in the place Bedrock Nova expects, added sample test to repro to the original report here:

https://github.com/BerriAI/litellm/issues/7087#issuecomment-2543378690

**Comment by ghost** [comment]
Closing as this is now supported - https://github.com/BerriAI/litellm/blob/0dbf71291e9ed261a15ada370668c3a6598ed990/litellm/llms/bedrock/chat/converse_transformation.py#L338

**Comment by ghost** [comment]
Thank you for the initial pr here @ryanh-ai

**Comment by ryanh-ai** [comment]
@krrishdholakia  I'm still having an issue with this, repro is here:

https://github.com/BerriAI/litellm/issues/7087#issuecomment-2543378690

**Pull request closed without merging**